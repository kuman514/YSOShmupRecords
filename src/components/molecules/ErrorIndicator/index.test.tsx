import React from 'react';
import { beforeEach, describe, expect, it } from 'vitest';
import { cleanup, render } from '@testing-library/react';

import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { ErrorIndicator } from '.';

describe('ErrorIndicator', () => {
  beforeEach(() => {
    cleanup();
  });

  it('should have a snapshot match for its appearance', () => {
    const routes = [
      {
        path: '/',
        element: <ErrorIndicator title="오류가 발생했습니다." />,
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ['/'],
      initialIndex: 0,
    });

    const { container } = render(<RouterProvider router={router} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
