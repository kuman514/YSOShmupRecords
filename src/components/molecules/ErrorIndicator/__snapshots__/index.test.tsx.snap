// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorIndicator > should have a snapshot match for its appearance 1`] = `
<div
  class="sc-grXZZQ fpEzvy"
>
  <img
    alt="에러 발생"
    class="sc-dkjKgF fXTRgv"
    src="/src/assets/images/error.png"
  />
  <h2
    class="sc-hLclGa jPMdaH"
  >
    오류가 발생했습니다.
  </h2>
  <span
    class="sc-eDnVMP icsQxT"
  >
    인터넷 연결 상태와 현재 주소를 확인해주세요.
  </span>
  <span
    class="sc-eDnVMP icsQxT"
  >
    서버 상의 에러일 수 있으니 잠시 후 다시 시도해주세요.
  </span>
  <span
    class="sc-eDnVMP icsQxT"
  >
    문제 발생 시 아래 카카오 오픈채팅방 버튼으로 문의해주세요.
  </span>
  <button
    class="sc-bcPKhP EfXTN"
  >
    타이틀 화면으로 돌아가기
  </button>
  <button
    class="sc-bcPKhP EfXTN"
  >
    소통창구 (카카오 오픈채팅방)
  </button>
</div>
`;
